### PR TITLE
Kseq

### PR DESCRIPTION
**Pull request by cosunae**
## Technical Description

Allows code generation for icosahedral backend to generate solvers with forward or backward loop order

### Example & Testing

Tested with two (forward & backward) solver examples in icondusk-e2e
https://github.com/dawn-ico/icondusk-e2e/pull/39

Tests are implemented for naive & cuda and results are checked against standalone C++ solution.

### CODE REVIEW

**Comment by muellch** [comment]
launch jenkins

File: dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp [hunk 1]
```diff
@@ -1530,7 +1583,8 @@ generateF90InterfaceSI(FortranInterfaceModuleGen& fimGen,
     }
     if(includeErrorThreshold) {
       wrapper.addBodyLine(
-          "   MERGE(rel_err_threshold,DEFAULT_RELATIVE_ERROR_THRESHOLD,present(rel_err_threshold)) "
+          "   "
+          "MERGE(rel_err_threshold,DEFAULT_RELATIVE_ERROR_THRESHOLD,present(rel_err_threshold)) "
```
Comment (Stagno) [inline dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp:1587]: Be sure to pull changes from current master, as the `MERGE` way to do select the threshold is removed there (for good reasons).

**Comment by muellch** [comment]
launch jenkins

File: dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp (hunk 1 above)
Comment (cosunae) [reply dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp]: done, thanks

**Comment by cosunae** [comment]
launch jenkins

**Review by muellch — APPROVED** [review]
Looks good!

**Pull request merged**